vowpalwabbit/io_to_queue.h

```diff
@@ -0,0 +1,76 @@
+#ifndef _IO_QUEUE_
+#define _IO_QUEUE_
+
+#include <thread>
+#include <future>
+#include <mutex>  
+#include <chrono>   
+
+#include "io_item.h"
+
+//Adds a line of input to the input queue
+inline bool add_to_queue(vw& all, char *& line){
+
+  bool finish = false;
+
+  size_t num_chars_initial = readto(*(all.p->input), line, '\n');
+  
+  if(num_chars_initial < 1 || strlen(line) < 1){
+      finish = true;
+  }
+
+  IO_Item line_item(std::string(line), num_chars_initial);
```

Comment:
Rather than constructing this and then copying it on line 26, you can either do a move or better yet just emplace this item directly on line 26.